transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/connector/ControlIT.java

```diff
@@ -126,21 +109,13 @@ public Object invoke(Invocation invocation) throws Throwable {
         "server.send.close.payload.length.126/handshake.response.and.frame" })
     public void shouldFailWebSocketConnectionWhenServerSendCloseFrameWithPayloadTooLong() throws Exception {
         final IoHandler handler = context.mock(IoHandler.class);
-        final CountDownLatch latch = new CountDownLatch(1);
 
         context.checking(new Expectations() {
             {
                 oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
                 oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
-                oneOf(handler).exceptionCaught(with(any(IoSessionEx.class)), with(any(Throwable.class)));
-                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
-                will(new CustomAction("Latch countdown") {
-                    @Override
-                    public Object invoke(Invocation invocation) throws Throwable {
-                        latch.countDown();
-                        return null;
-                    }
-                });
+                allowing(handler).exceptionCaught(with(any(IoSessionEx.class)), with(any(Exception.class)));
+                allowing(handler).sessionClosed(with(any(IoSessionEx.class)));
```

Comment:
I don't think we need to allow session closed?  It should be deterministic when the tcp connection will be closed based on the chain rule,  as context is the most inner in this case it should always be open at that point.  I would believe